ilastik/plugins_default/tracking_csv_export.py

```diff
@@ -0,0 +1,96 @@
+        excludedFeatures = ['Histogram']
+        
+        # check which features are present and construct table of the appropriate size
+        frame, _ = graph.nodes_iter().next()
+        for category in features[frame].keys():
+            for feature in features[frame][category].keys():
+                if feature not in excludedFeatures:
+                    if (np.array(features[frame][category][feature])).ndim == 2:
+                        for column in range(np.array(features[frame][category][feature]).shape[1]):
+                            singleFeatureValueName = '{f}_{c}'.format(f=feature, c=column)
+                            headers.append(singleFeatureValueName)
+                    else:
+                        headers.append(feature)
+
+        table = np.zeros([graph.number_of_nodes(), len(headers)])
+
+        for rowIdx, node in enumerate(graph.nodes_iter()):
+            frame, label = node    
+            trackId = graph.node[node]['trackId']
+            lineageId = graph.node[node]['lineageId']
+
+            table[rowIdx, 0] = frame
+            table[rowIdx, 1] = label
+            table[rowIdx, 2] = trackId
+            table[rowIdx, 3] = lineageId
+
+            # insert parent of a division
+            try:
+                table[rowIdx, 4] = graph.node[graph.node[node]['parent']]['trackId']
+            except KeyError:
+                table[rowIdx, 4] = 0
+
+            # insert merger
+            try:
+                if isinstance(graph.node[node]['mergerValue'], int):
+                    table[rowIdx, 5] = graph.node[node]['mergerValue']
+                else:
+                    table[rowIdx, 5] = 0
+            except KeyError:
+                table[rowIdx, 5] = 0
+            colIdx = 6
+
+            for category in features[frame].keys():
+                for feature in features[frame][category].keys():
+
+                    if feature not in excludedFeatures:
+                        if (np.array(features[frame][category][feature])).ndim == 2:
+                            for column in range(np.array(features[frame][category][feature]).shape[1]):
+                                try:
+                                    table[rowIdx, colIdx] = features[frame][category][feature][label, column]
+                                except IndexError:
+                                    if 'SquaredDistances' in feature:
+                                        table[rowIdx, colIdx] = 9999
+                                    else:
+                                        table[rowIdx, colIdx] = 0
+                                colIdx += 1
+                        else:
+                            table[rowIdx, colIdx] = features[frame][category][feature][label]
+                            colIdx += 1
+
+        # sort table by frame, then labelImage
+        table = table[table[:,1].argsort()]
+        table = table[table[:,0].argsort(kind='mergesort')]
```

Comment:
Ah, I see.

>AFAIK numpy doesn't provide a simple way to sort by two keys.

You can do it with [np.lexsort()](https://docs.scipy.org/doc/numpy/reference/generated/numpy.lexsort.html).

```python
# sort table by frame, then labelImage
table = table[ np.lexsort(table[:,:2].transpose()[::-1]) ]
```

(The `[::-1]` is necessary because lexsort considers the *last* position to be the primary key.)